src/runtime/metal.cpp

```diff
@@ -789,6 +789,22 @@ WEAK int halide_metal_run(void *user_context,
 
     set_compute_pipeline_state(encoder, pipeline_state);
 
+    int num_kernel_args = 0;
+    for (int i = 0; args[i] != nullptr; i++) {
+        ++num_kernel_args;
+    }
+
+    // NOTE(slomp): arg_sizes[] must terminate with a zero!
+    size_t *arg_sizes = (size_t *)__builtin_alloca((num_kernel_args + 1) * sizeof(size_t));
+    for (int i = 0; i < num_kernel_args; i++) {
+        if (arg_is_buffer[i]) {
+            arg_sizes[i] = sizeof(void *);  // or should it always be uint64_t as in line 876 below?
```

Comment:
In practice, sizeof(void*) == sizeof(uint64_t) for all Metal applications since all Metal OS versions are 64-bitonly.

Reply:
It looks like we are generating 32bits metal modules, though:
```
1>Generating initmod.metal_32.bc
1>Generating initmod.metal_32_debug.bc
```